src/gui/Src/Gui/PatchDialog.cpp

```diff
@@ -682,8 +684,228 @@ void PatchDialog::saveAs1337(const QString & filename)
 
     QFile file(filename);
     file.open(QFile::WriteOnly | QFile::Text);
-    QString text = lines.join("\n");
-    file.write(text.toUtf8().constData(), text.length());
+    if(file.error() != QFile::NoError)
+    {
+        QMessageBox msg(QMessageBox::Critical, tr("Error"), tr("Cannot export patches because an error occurs during file I/O"));
+        msg.exec();
+        return;
+    }
+    QString text = lines.join("\r\n");
+    QByteArray utf8Data = text.toUtf8();
+    file.write(utf8Data.constData(), utf8Data.length());
+    file.close();
+
+    QMessageBox msg(QMessageBox::Information, tr("Information"), tr("%1 patch(es) exported!").arg(patches));
+    msg.setWindowIcon(QIcon(":/icons/images/information.png"));
+    msg.setParent(this, Qt::Dialog);
+    msg.setWindowFlags(msg.windowFlags() & (~Qt::WindowContextHelpButtonHint));
+    msg.exec();
+}
+
+void PatchDialog::saveAsC(const QString &filename)
+{
+    QString template_C;
+    // Loading template
+    template_C = QString(R"xxx(

```

Comment:
Is this supported by Visual Studio 2013? My preference would be to load templates from a file that is not contained in the x64dbg source code directly (we can create a separate repository for it and include it as a submodule). I also think there should be a separate button for saving with a template so the current behavior is unchanged.